src/config.ts

```diff
@@ -0,0 +1,64 @@
+import fs from 'fs'
+import { resolve } from 'path'
+import { ResolvedConfig } from 'vite'
+import { GenerateSWConfig, InjectManifestConfig } from 'workbox-build'
+import { ManifestOptions, VitePWAOptions, ResolvedVitePWAOptions } from './types'
+import { cachePreset } from './cache'
+
+export function resolveOptions(options: Partial<VitePWAOptions>, viteConfig: ResolvedConfig): ResolvedVitePWAOptions {
+  const root = viteConfig.root
+  const pkg = fs.existsSync('package.json')
+    ? JSON.parse(fs.readFileSync('package.json', 'utf-8'))
+    : {}
+  const {
+    srcDir = options.srcDir || 'public',
+    outDir = options.outDir || viteConfig.build.outDir || 'dist',
+    filename = options.filename || 'sw.js',
+    strategies = options.strategies || 'generateSW',
+  } = options
```

Comment:
```suggestion
  const {
    srcDir = 'public',
    outDir = viteConfig.build.outDir || 'dist',
    filename = 'sw.js',
    strategies = 'generateSW',
  } = options
```

This should be enough to serve default values.